main/src/ca/uwaterloo/flix/language/phase/Monomorpher.scala

```diff
@@ -114,6 +114,7 @@ object Monomorpher {
       case Kind.RecordRow => Type.RecordRowEmpty
       case Kind.SchemaRow => Type.SchemaRowEmpty
       case Kind.Predicate => Type.mkAnyType(tpe0.loc)
+      case Kind.JvmConstructorOrMethod => Type.mkAnyType(tpe0.loc) // to check
```

Comment:
this should throw: InternalCompilerException("Unexpected kind: 'JvmConstructorOrMethod'").

Reply:
And throwing cases are always last.

Reply:
This remains to be done :)